react/components/molecules/input-form/input-form.js

```diff
@@ -0,0 +1,151 @@
+import React, { PureComponent } from "react";
+import { StyleSheet, Text, ViewPropTypes } from "react-native";
+import PropTypes from "prop-types";
+import { mix } from "yonius";
+
+import { IdentifiableMixin, baseStyles } from "../../../util";
+
+import { Input, TextArea, Touchable } from "../../atoms";
+
+export class InputForm extends mix(PureComponent).with(IdentifiableMixin) {
+    static get propTypes() {
+        return {
+            label: PropTypes.string,
+            placeholder: PropTypes.string,
+            value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+            meta: PropTypes.string,
+            secureTextEntry: PropTypes.bool,
+            onValueUpdate: PropTypes.func,
+            onBlur: PropTypes.func,
+            onFocus: PropTypes.func,
+            style: ViewPropTypes.style
+        };
+    }
+
+    static get defaultProps() {
+        return {
+            label: undefined,
+            placeholder: undefined,
+            value: undefined,
+            meta: undefined,
+            secureTextEntry: false,
+            onValueUpdate: value => {},
+            onFocus: () => {},
+            onBlur: () => {},
+            style: {}
+        };
+    }
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            valueData: this.props.value
+        };
+    }
+
+    componentDidUpdate(prevProps) {
+        if (prevProps.value !== this.props.value) {
+            this.setState(
+                {
+                    valueData: this.props.value
+                },
+                () => {
+                    this.focus();
+                }
+            );
+        }
+    }
+
+    _onFocus = () => {
+        this.props.onFocus();
+    };
+
+    _onBlur = () => {
+        this.props.onBlur();
+    };
+
+    blur = () => {
+        this.input?.blur();
+    };
+
+    focus = () => {
+        this.input?.focus();
+    };
+
+    onHeaderPress = () => {
+        this.focus();
+    };
+
+    onChangeValue = value => {
+        this.setState({ valueData: value }, () => {
+            this.props.onValueUpdate(value);
+        });
+    };
+
+    _style = () => {
+        return [styles.inputForm, this.props.style];
+    };
+
+    _renderInput = () => {

```

Comment:
could it be better if instead of rendering an actual input/textArea we just leave for the user to decide using slots l[ike I'm doing here to override the card content](https://github.com/ripe-tech/ripe-components-react-native/compare/dva/card?expand=1#diff-c2679bdcf3b2a154cc180d396f7d6aa358a86657a11145c96c668a8705522654R107), this would also save a lot of props
not sure if in terms of style it would be possible to integrate it as easily but I think it's worth a try for the simplification it would do

Reply:
You are saying maintain this logic but also add the possibility to override it with the children? Or remove the default logic?

Reply:
I was saying to remove the default logic but only if you can maintain the same style. It makes sense to keep it this way if we always have to override the style to make it look good

Reply:
But we can't maintain the style. To maintain the style, the parent component must know to set a specific height to the input and to remove the background color and padding to the textarea. I made it like this (diverging from components-vue form-input) to maintain consistency in the styling of the component and allocate all those decisions in one place.